client/lib/signup/step-actions/index.js

```diff
@@ -227,6 +228,23 @@ function getNewSiteParams( {
 		newSiteParams.options.in_page_builder = true;
 	}
 
+	if ( selectedDesign ) {
+		// If there's a selected design, use similar new site options to what used
+		// in gutenboarding (see `client/landing/gutenboarding/stores/onboard/actions.ts`)
+		// `Design` type defined in `packages/design-picker/src/types.ts`
+		newSiteParams.options.theme = `pub/${ selectedDesign.theme }`;
+		newSiteParams.options.template = selectedDesign.template;
+		newSiteParams.options.font_base = selectedDesign.fonts.base;
+		newSiteParams.options.font_headings = selectedDesign.fonts.headings;
+		newSiteParams.options.use_patterns = true;
+
+		// TODO: needs same value to be supported in the backend
+		//(see https://github.com/Automattic/wp-calypso/issues/51231)
+		// Is this override necessary? Before overriding, the site_creation_flow
+		// is "with-design-picker"
+		newSiteParams.options.site_creation_flow = 'onboarding-with-design';
```

Comment:
Is this override necessary? Before overriding, the site_creation_flow is "with-design-picker"

Reply:
Good point! We should probably use the same flow value regardless of the design selection (including no selection/ skip). Then we should take this into consideration when implementing https://github.com/Automattic/wp-calypso/issues/51231 so a 'gutenboarding-like' site to be created only when `template` and `fonts` are passed as well 🤔

Reply:
Backend check in #51231 could become:

`if (isGutenboarding || isSiteWithThemeFontsAndBlocks()) { ... }`